src/OpenTelemetry/Metrics/Exemplar/AllExemplarFilter.cs

```diff
@@ -0,0 +1,32 @@
+// <copyright file="AllExemplarFilter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Metrics
+{
+    /// <summary>
+    /// The Exemplar Filter which always samples measurements./>
+    /// </summary>
+    internal sealed class AllExemplarFilter : IExemplarFilter
+    {
+        public bool ShouldSample(long value, ReadOnlySpan<KeyValuePair<string, object>> tags)
```

Comment:
The value has a `long` type, what's the current thinking of other types (e.g. double)?

Reply:
Will we only ever have `long` & `double`? Just wondering if it should be `ShouldSample<T>(T value, ReadOnlySpan<KeyValuePair<string, object>> tags)`.

Reply:
at the earliest stage of callback from .NET runtime, the SDK converts everything to long or double.
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/MeterProviderSdk.cs#L262-L270

Reply:
in the future where we allow pluggable aggregators, they would also see long/double (per current design...)